### PR TITLE
add UnitLowerCholeskyTransform; change default parameterization for AutoMultivariateNormal

### PR DESCRIPTION
**Pull request by martinjankowiak**
addresses #2963 

questions for reviewers:
- did i put things in the right files?
- do i need additional tests?
- do i need to change any additional autoguide defaults?
- do i need to do anything on the doc side of things for `unit_lower_cholesky`?

### CODE REVIEW

File: pyro/distributions/transforms/unit_cholesky.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    codomain = unit_lower_cholesky
+
+    def __eq__(self, other):
+        return isinstance(other, UnitLowerCholeskyTransform)
+
+    def _call(self, x):
+        return x.tril(-1) + torch.eye(x.size(-1), device=x.device, dtype=x.dtype)
+
+    def _inverse(self, y):
+        return y.tril(-1)
```
Comment (fritzo) [inline pyro/distributions/transforms/unit_cholesky.py:27]: It's cheaper to simply `return y`.

File: docs/source/distributions.rst [hunk 2]
```diff
@@ -522,6 +522,13 @@ SoftplusLowerCholeskyTransform
     :undoc-members:
     :show-inheritance:
 
+UnitLowerCholeskyTransform
```
Comment (fritzo) [inline docs/source/distributions.rst:525]: It looks like we're trying to keep this file alphabetical, can you try to preserve that convention?

**Review by fritzo — CHANGES_REQUESTED** [review]
LGTM except for minor comments.

> did i put things in the right files?
:+1: yes

> do i need additional tests?
Tests already look sufficient.

> do i need to change any additional autoguide defaults?
Nope, it looks like `AutoMultivariateNormal` is the only use of `lower_cholesky`.

> do i need to do anything on the doc side of things for unit_lower_cholesky?
I don't think so, but can you `make docs` and check that `unit_lower_cholesky` shows up?

File: pyro/distributions/transforms/unit_cholesky.py (hunk 1 above)
Comment (martinjankowiak) [reply pyro/distributions/transforms/unit_cholesky.py:27]: done

File: docs/source/distributions.rst (hunk 2 above)
Comment (martinjankowiak) [reply docs/source/distributions.rst:525]: done

**Comment by martinjankowiak** [comment]
unfortunately i'm unable to build the docs:

```
Warning, treated as error:
autodoc: failed to import module 'multi_mnist' from module 'pyro.contrib.examples'; the following exception was raised:
Traceback (most recent call last):
.........
  File "/Users/mjankowi/pyro/pyro/contrib/examples/util.py", line 15, in MNIST
    ] + datasets.MNIST.mirrors
AttributeError: type object 'MNIST' has no attribute 'mirrors'
```

tried to upgrade sphinx etc but that didn't help...

**Comment by fritzo** [comment]
> i'm unable to build the docs:

Try upgrading torchvision?
```
pip install -U torch torchvision
```

**Comment by martinjankowiak** [comment]
ah yeah torchvision was it. and with the addition to `__all__` i can confirm the docs build and the appropriate entry shows up

**Review by fritzo — APPROVED** [review]
:+1: Thanks for checking docs!

**Pull request merged**